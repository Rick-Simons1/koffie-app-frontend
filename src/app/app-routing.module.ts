import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { CoffeeListComponent } from '../app/coffee-list/coffee-list.component';
import { CoffeeFormComponent } from '../app/coffee-form/coffee-form.component';
import { CoffeeEditFormComponent } from './coffee-edit-form/coffee-edit-form.component';
import { CoffeeOrderPageComponent } from './coffee-order-page/coffee-order-page.component';
import { OrderDetailListComponent } from './order-detail-list/order-detail-list.component';
import { RegisterPageComponent } from './register-page/register-page.component';
import { LoginPageComponent } from './login-page/login-page.component';
import { UserPageComponent } from './user-page/user-page.component';
import { UserEditFormComponent } from './user-edit-form/user-edit-form.component';
import { UserEditPasswdComponent } from './user-edit-passwd/user-edit-passwd.component';



const routes: Routes = [
  { path: 'coffees', component: CoffeeListComponent },
  { path: 'addCoffee', component: CoffeeFormComponent },
  { path: 'editCoffee', component: CoffeeEditFormComponent },
  { path: 'orderCoffee', component: CoffeeOrderPageComponent },
  { path: 'orderDetails', component: OrderDetailListComponent },
  { path: 'register', component: RegisterPageComponent },
  { path: 'login', component: LoginPageComponent },
  { path: 'account', component: UserPageComponent },
  { path: 'account/edit', component: UserEditFormComponent },
  { path: 'account/edit/password', component: UserEditPasswdComponent }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
