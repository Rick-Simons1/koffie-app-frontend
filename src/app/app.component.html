<div class="container">
  <div class="row">
    <div class="col-md-12">
      <div class="card bg-dark my-5">
        <div class="card-body">
          <h2 class="card-title text-center text-white py-3">{{ title }}</h2>
          <ul class="text-center list-inline py-3">
            <li class="list-inline-item">
              <a routerLink="/login" *ngIf="!loginService.loggedIn" class="btn btn-info">login</a>
            </li>
            <li class="list-inline-item">
              <a routerLink="/register" *ngIf="!loginService.loggedIn" class="btn btn-info">register</a>
            </li>
            <li class="list-inline-item">
              <a routerLink="/login" *ngIf="loginService.loggedIn" class="btn btn-info" (click)="logout()">logout</a>
            </li>
            <li class="list-inline-item">
              <a routerLink="/coffees" *ngIf="(loginService.loggedIn && loginService.admin)" class="btn btn-info">List coffees</a>
            </li>
            <li class="list-inline-item">
              <a routerLink="/addCoffee" *ngIf="(loginService.loggedIn && loginService.admin)" class="btn btn-info">Add coffee</a>
            </li>
            <li class="list-inline-item">
              <a routerLink="/orderCoffee" *ngIf="loginService.loggedIn" class="btn btn-info">order coffee</a>
            </li>
            <li class="list-inline-item">
              <a routerLink="/orderDetails" *ngIf="loginService.loggedIn" class="btn btn-info">make order</a>
            </li>
            <li class="list-inline-item">
              <a routerLink="/account" *ngIf="loginService.loggedIn" class="btn btn-info">account page</a>
            </li>

          </ul>
        </div>
      </div>
      <router-outlet></router-outlet>
    </div>
  </div>
</div>
