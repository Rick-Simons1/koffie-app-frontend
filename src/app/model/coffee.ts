export class Coffee {
  id: number;
  coffeeName: string;
  coffeeDescription: string;
  coffeePrice: number
}
