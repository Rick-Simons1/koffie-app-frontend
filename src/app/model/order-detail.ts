import { Order } from './order';

export class OrderDetail {
  orderDetailId: number;
  order: Order;
  coffeeId: number;
  coffeeName: string;
  amountOfSugar: number;
  amountOfMilk: number;
  coffeeCode: string;
  placedByUserId: string;
}
