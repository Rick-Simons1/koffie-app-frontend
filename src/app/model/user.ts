export class User {
  id: string;
  username: string;
  password: string;
  amountOfOrdersDelivered: number;
  amountOfCoffeesOrdered: number;
  name: string;
}
