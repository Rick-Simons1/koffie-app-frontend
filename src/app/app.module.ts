import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';
import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { CoffeeListComponent } from './coffee-list/coffee-list.component';
import { CoffeeFormComponent } from './coffee-form/coffee-form.component';
import { CoffeeService } from './service/coffee.service';
import { CoffeeEditFormComponent } from './coffee-edit-form/coffee-edit-form.component';
import { AlertModule } from './_alert';
import { CoffeeOrderPageComponent, CoffeeSelectComponent } from './coffee-order-page/coffee-order-page.component';
import { MatDialogModule } from '@angular/material/dialog';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { OrderdetailService } from './service/orderdetail.service';
import { OrderDetailListComponent } from './order-detail-list/order-detail-list.component';
import { RegisterPageComponent } from './register-page/register-page.component';
import { LoginPageComponent } from './login-page/login-page.component';
import { UserPageComponent } from './user-page/user-page.component';
import { UserEditFormComponent } from './user-edit-form/user-edit-form.component';
import { UserEditPasswdComponent } from './user-edit-passwd/user-edit-passwd.component';

@NgModule({
  declarations: [
    AppComponent,
    CoffeeListComponent,
    CoffeeFormComponent,
    CoffeeEditFormComponent,
    CoffeeOrderPageComponent,
    CoffeeSelectComponent,
    OrderDetailListComponent,
    RegisterPageComponent,
    LoginPageComponent,
    UserPageComponent,
    UserEditFormComponent,
    UserEditPasswdComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
    FormsModule,
    AlertModule,
    MatDialogModule,
    BrowserAnimationsModule
    
  ],
  providers: [CoffeeService, OrderdetailService],
  bootstrap: [AppComponent]
})
export class AppModule { }
