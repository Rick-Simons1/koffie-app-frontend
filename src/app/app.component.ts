import { Component } from '@angular/core';
import { UserService } from './service/user.service';
import { User } from './model/user';
import { LoginServiceService } from './service/login-service.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title: String;

  
  user: User;

  constructor(private userService: UserService, public loginService: LoginServiceService) {
    this.title = "Koffie-app"
    this.user = new User;
    loginService.admin = true;
  }

  ngOnInit(): void {
    if (this.loginService.loggedIn == false) {
      localStorage.removeItem("token");
      console.log("removed");
    }
  }



  logout() {
    this.loginService.loggedIn = false;
    localStorage.removeItem("token");
    console.log("loggedout");
  }
}
