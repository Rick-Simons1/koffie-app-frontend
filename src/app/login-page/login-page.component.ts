import { Component, OnInit } from '@angular/core';
import { User } from '../model/user';
import { UserService } from '../service/user.service';
import { HttpResponse } from '@angular/common/http';
import { Router } from '@angular/router';
import { LoginServiceService } from '../service/login-service.service';
import { error } from '@angular/compiler/src/util';
import { AlertModule, AlertService } from '../_alert';

@Component({
  selector: 'app-login-page',
  templateUrl: './login-page.component.html',
  styleUrls: ['./login-page.component.css']
})
export class LoginPageComponent implements OnInit {

  user: User;

  constructor(private userService: UserService, private router: Router, private loginService: LoginServiceService, private alertService: AlertService) {
    this.user = new User();
  }

  ngOnInit(): void {
  }

  onSubmit() {
    this.userService.loginUser(this.user).subscribe(response => {
      const token = response.headers.get("Authorization");
      localStorage.setItem("token", token);
      this.router.navigate(['/account'])
      this.loginService.loggedIn = true;
      if (this.user.username == "admin") {
        this.loginService.admin = true;
      }
      else {
        this.loginService.admin = false;
      }
    }, error => this.handleError(error))
  }

  handleError(error: any) {
    if (error.status == 403) {
      this.sendAlertMsg(error.status)
    }
  }

  sendAlertMsg(result: number) {
    var msg = "";
    if (result === 403) {
      msg = "login credentials are wrong please try again";
      this.alertService.error(msg);
      return msg;
    }
    else {
      msg = "login succesfull";
      this.alertService.success(msg);
      return msg;
    }
  }

}
