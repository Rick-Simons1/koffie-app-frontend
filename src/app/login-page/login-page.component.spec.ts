import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { LoginPageComponent } from './login-page.component';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { User } from '../model/user';
import { UserService } from '../service/user.service';
import { HttpResponse } from '@angular/common/http';
import { Router } from '@angular/router';
import { AppComponent } from '../app.component';
import { RouterTestingModule } from '@angular/router/testing';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { AlertModule } from '../_alert';
import { browser } from 'protractor';

describe('LoginPageComponent', () => {
  let component: LoginPageComponent;
  let fixture: ComponentFixture<LoginPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [LoginPageComponent],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
        FormsModule,
        ReactiveFormsModule,
        AlertModule
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('sould send error msg when credntials are wrong', () => {
    expect(component.sendAlertMsg(403)).toBe("login credentials are wrong please try again")
  })
  it('sould send succes msg when credntials are right', () => {
    expect(component.sendAlertMsg(200)).toBe("login succesfull")
  })

  it('test', () => {
    var user: User = new User();
    user.username = "admin";
    user.password = "password";
    component.user = user
    expect(fixture).toMatchSnapshot();
    
  });
});

describe('loginpage gui tests', () => {
  it('redirect to account', () => {
    
  })
});


