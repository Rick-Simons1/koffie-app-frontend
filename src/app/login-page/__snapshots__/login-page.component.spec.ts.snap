// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPageComponent test 1`] = `
<app-login-page
  alertService={[Function AlertService]}
  loginService={[Function LoginServiceService]}
  router={[Function Router]}
  user={[Function User]}
  userService={[Function UserService]}
>
  <div
    class="card my-5"
  >
    <div
      class="card-body"
    >
      <alert>
        
      </alert>
      <form
        class="ng-untouched ng-pristine ng-valid"
        novalidate=""
      >
        <div
          class="form-group"
        >
          <label
            for="username"
          >
            username
          </label>
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="username"
            name="username"
            placeholder="Enter your username"
            required=""
            type="text"
          />
        </div>
        <div
          class="alert alert-danger"
        >
          username is required
        </div>
        <div
          class="form-group"
        >
          <label
            for="password"
          >
            password
          </label>
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            id="password"
            name="password"
            placeholder="Enter the password you want to use"
            required=""
            type="text"
          />
          <div
            class="alert alert-danger"
          >
            the password is required
          </div>
        </div>
        <button
          class="btn btn-info"
          type="submit"
        >
           Submit 
        </button>
      </form>
    </div>
  </div>
</app-login-page>
`;
